packages/react-sdk/src/components/PendingCallPanel/PendingCallPanel.tsx

```diff
@@ -21,24 +21,43 @@ const CALLING_STATE_TO_LABEL: Record<CallingState, string> = {
   [CallingState.LEFT]: 'Left call',
 };
 
-export const PendingCallPanel = () => {
+export type PendingCallPanelProps = {
+  /**
+   * Whether to include the current user in the list of members to show.
+   * @default false.
+   */
+  includeSelf?: boolean;
+
+  /**
+   * The maximum number of members to show.
+   * @default 3.
+   */
+  totalMembersToShow?: number;
+};
+
+export const PendingCallPanel = (props: PendingCallPanelProps) => {
+  const { includeSelf = false, totalMembersToShow = 3 } = props;
   const call = useCall();
   const callingState = useCallCallingState();
   const { t } = useI18n();
-  const metadata = useCallMetadata();
   const members = useCallMembers();
+  const connectedUser = useConnectedUser();
 
   if (!call) return null;
 
-  const caller = metadata?.created_by;
-  const membersToShow = call.isCreatedByMe
-    ? members
-        ?.slice(0, 3)
-        .map(({ user }) => user)
-        .filter((u) => !!u) || []
-    : caller
-    ? [caller]
-    : [];
+  const membersToShow: UserResponse[] = (members || [])
+    .slice(0, totalMembersToShow - (includeSelf ? 1 : 0))
+    .map(({ user }) => user)
+    .filter((user) => user.id !== connectedUser?.id || includeSelf);
+  if (
```

Comment:
This code is a bit hard to follow. Could it be simplified?

Reply:
I'll give it one more round :)